spec/requests/vendor_api/confirm_enrolment_spec.rb

```diff
@@ -66,4 +66,18 @@
     expect(response).to have_http_status(422)
     expect(parsed_response).to be_valid_against_openapi_schema('BadRequestBodyResponse')
   end
+
+  it 'returns an error if the request is not authenticated' do
+    post_api_request '/api/v1/applications/not-important/confirm-enrolment',
+                     headers: { "Authorization": 'invalid-token' }
+
+    expect(response).to have_http_status(401)
+  end
+
+  it 'returns an error if the no API token is present' do
+    post_api_request '/api/v1/applications/not-important/confirm-enrolment',
+                     headers: { "Authorization": nil }
```

Comment:
What if the header is missing altogether? It's nice to show a useful message in that case. (I appreciate `deep_merge` doesn't make this convenient rn)

Reply:
Added test for this in the latest iteration.